### DISCUSSION TITLE
Method for Copilot to read reference files (yaml)

### DISCUSSION THREAD

**Discussion by macaws** [community/community#7651] [Copilot Conversations]
I'm writing a new API and this is where I can see Copilot being the most helpful due to how repetitious they are. I have an OpenAPI reference file in the root project directory, but it doesn't appear to be inferring anything from it.

I tried pasting the YAML file into a TS file commented out and imported it, but that doesn't seem to help it either. Is there a way to do this, or is this something that we might be able to see in future releases?

**Comment by MulleDK19** [comment]
Copilot only knows about what's in the current file, and only what's behind the caret. Some languages will have some context injected to know about functions after the caret, but even then it won't know about their contents.

This is a current shortcoming of GPT-3 in general, and I'm not sure how they're gonna fix it, if at all.

**Reply by macaws** [reply]
This definitely is a shortcoming, but more so that when using TS it doesn't recognize classes what I write outside of the file. It makes suggestions that are not valid with the class structure. At times like this, it's more in my way than helpful.

**Reply by Urethramancer** [reply]
Project-wide awareness is something we've all asked for, and I think they've promised it in the future. I just wish that future was closer :)

I'm also writing an API now, and it's annoying to not have it get the structure before I've "warmed up" the current file with one or two functions. But with only a few verbs, I'm basically writing/pasting most of it myself. At least it's gotten better at figuring out the trickier SQL for me now.

**Comment by inostia** [comment]
Is this something that's going to be possible in the future? It'd be nice to be able to make a comment like `# This file is modeled after {insert_file_here.ext}` and have Co-pilot follow the links to do the right thing.
